worktree_test.go

```diff
@@ -1358,6 +1358,24 @@ func (s *WorktreeSuite) TestResetSparsely() {
 	s.Equal("crappy.php", files[0].Name())
 }
 
+func (s *WorktreeSuite) TestResetSparselyInvalidDir() {
+	fs := memfs.New()
+	w := &Worktree{
+		r:          s.Repository,
+		Filesystem: fs,
+	}
+
+	testcases := [][]string{
+		{"non-existent"},
+		{"php/crappy.php"}, // non-dir
```

Comment:
Let's expand this to be more like table driven tests, so that we can have different input/output checks. With that, please add a test case ensuring the behaviour of `SkipSparseDirValidation`.